pynamodb/connection/base.py

```diff
@@ -783,14 +783,14 @@ def update_item(self,
             value = update.get(VALUE)
             attr_type, value = self.parse_attribute(value, return_type=True)
             action = update.get(ACTION)
-            if attr_type is None and action is not None and action.upper() != DELETE:
+            if attr_type is None and action is not None and (action.upper() != DELETE or (value is not None and not(isinstance(value, str) and value.upper() == 'NULL'))):
```

Comment:
Can you extract this condition to a function? It's being repeated in multiple places.